transformers/events/DOCUMENTATION.md

```diff
@@ -2,116 +2,206 @@
+            * Open the 'Contract' tab and in the contract source code and search
+            for the `LogItemUpdate` event. It looks like this: `event LogItemUpdate(uint id);`. 
+            * You can copy that string and paste it into the Keccak-256
+            calculator (link above). Remove the parameter names and use
+            `uint256` for the `unit` so like this: `LogItemUpdate(uint256)`.
+            This will output an event signature.
+            * To verify you got the signature right you can take that signature
+            (in this case
+            `a2c251311b1a7a475913900a2a73dc9789a21b04bc737e050bbc506dd4eb3488`) 
+            and search for it on the events tab of the contract. Make sure you 
+            prefix it with `0x`. The results should include the event you are
+            looking for, provided one already exists for it.
+    1. Make the test pass by updating the function list in `signature.go` as
+       well as updating `method.go` as needed. Both of these files are in the
+       constants package.
+    1. Create test event in [`test_data`](../test_data)
+        * Use Data from a real event in Etherscan wherever possible.
+          Particularly when it comes to Topics and the Data entries. You'll
+          thank me later.
+        * Update constants as needed. Look at older examples of test data for
+          inspiration.
+        * You do not need a database migration at this point.
+    1. Create a new directory in `events` named after your new transformer. This
+       will be the package your transformer is stored in.
+    1. Create three files - `transformer.go`, `transformer_test.go` and
+       `<package_name>_test.go` in the new directory. See the other transformers
+       for examples.
+    1. `<package_name>_test.go` lets `gingko` run tests in this directory. The
+       file should look something like this:
+       
+```go
+ package <package_name>_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestLogMinSell(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "<PackageName>  Suite")
+}
+```
+   1. Implement the toModels function in `transformer.go` with the appropriate
+   testing in `transformer_test.go` of course. This means converting the raw log 
+   into a go struct. 
+   1. For custom events you can convert each EventLog entry into an entity
+   using the function `contract.UnpackLog(&entity, "<EventLogName>", log.Log)`.
+   The entity is usually defined (by you) in a file called `entity.go` in the
+   same package as the transformer. See the transformer in
+   `transformers/events/log_make` for an example of this method.
+   1. After converting the log entry to an entity, look up all of it's foreign
+      keys from it's data and assign them to the entity.
+   1. From the entity create an InsertionModel, which you return.
+   1. For `LogNote` events you'll need to look at the method signature of the
```

Comment:
I wonder if it would be helpful to have directions for LogNote and directions for custom events as nested bullets to signify that the you need to do one or the other, not both?

Something like:
```
 1. Implement the ToModels function in `transformer.go`...
    - For custom events:
        - ...
    - For LogNoteEvents:
        - ...
```

Reply:
👍